ledger/execution/tx_coinbase.go

```diff
@@ -169,7 +169,10 @@ func RetrievePools(chain *blockchain.Chain, db database.Database, blockHeight ui
 
 			if eliteEdgeNodeVotes != nil {
 				if eliteEdgeNodeVotes.Block == guardianVotes.Block {
-					eliteEdgeNodePool = storeView.GetEliteEdgeNodePool()
+					eliteEdgeNodePool, err = ledger.GetEliteEdgeNodePoolOfLastCheckpoint(eliteEdgeNodeVotes.Block)
```

Comment:
Would it be possible that guardianPool and eliteEdgeNodePool are associated with different blocks? It shouldn't be since guradianVoteBlock must be a directly finalized checkpoint, and hence eliteEdgeNodePool returned by ledger.GetEliteEdgeNodePoolOfLastCheckpoint() should always be associated with the same block?